src/http/ldp/PutOperationHandler.ts

```diff
@@ -36,9 +41,23 @@ export class PutOperationHandler extends OperationHandler {
       this.logger.warn('PUT requests require the Content-Type header to be set');
       throw new BadRequestHttpError('PUT requests require the Content-Type header to be set');
     }
+
+    // https://github.com/solid/community-server/issues/1027#issuecomment-988664970
+    // PUT is not allowed on metadata
+    if (this.metaStrategy.isAuxiliaryIdentifier(operation.target)) {
+      throw new ConflictHttpError('Not allowed to create or edit metadata resources using PUT.');
+    }
+
+    const exists = await this.store.resourceExists(operation.target, operation.conditions);
+
+    // Not allowed performing PUT on an already existing Container
+    // See https://github.com/solid/community-server/issues/1027#issuecomment-1023371546
+    if (exists && isContainerPath(operation.target.path)) {
+      throw new ConflictHttpError('Not allowed to PUT on already existing containers.');
+    }
```

Comment:
If this is checked in setRepresentation, it is not possible to add pim:storage during the containerInitializer (as it updates an existing container using setRepresentation).
So yes it can be checked there, but the behaviour is then different.

Reply:
We probably want to change the initializer in the future so it writes metadata directly to the metadata resource instead of how it does it now. Then this wouldn't be an issue. But that would greatly expand this PR so please just add a comment here explaining that is why it is currently done like that (and that we want to change this in the future).

Reply:
I will add that comment then.